src/go/k8s/api/redpanda/v1alpha2/common.go

```diff
@@ -0,0 +1,181 @@
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// KafkaAPISpec configures client configuration settings for connecting to Redpanda brokers.
+type KafkaAPISpec struct {
+	// Specifies a list of broker addresses in the format <host>:<port>
+	Brokers []string `json:"brokers"`
+	// Defines TLS configuration settings for Redpanda clusters that have TLS enabled.
+	// +optional
+	TLS *CommonTLS `json:"tls,omitempty"`
+	// Defines authentication configuration settings for Redpanda clusters that have authentication enabled.
+	// +optional
+	SASL *KafkaSASL `json:"sasl,omitempty"`
+}
+
+// KafkaSASL configures credentials to connect to Redpanda cluster that has authentication enabled.
+type KafkaSASL struct {
+	// Specifies the username.
+	// +optional
+	Username string `json:"username,omitempty"`
+	// Specifies the password.
+	// +optional
+	Password SecretKeyRef `json:"passwordSecretRef,omitempty"`
+	// Specifies the SASL/SCRAM authentication mechanism.
+	Mechanism SASLMechanism `json:"mechanism"`
+	// +optional
+	OAUth KafkaSASLOAuthBearer `json:"oauth,omitempty"`
+	// +optional
+	GSSAPIConfig KafkaSASLGSSAPI `json:"gssapi,omitempty"`
+	// +optional
+	AWSMskIam KafkaSASLAWSMskIam `json:"awsMskIam,omitempty"`
+}
+
+type SASLMechanism string
+
+const (
+	SASLMechanismPlain                  SASLMechanism = config.SASLMechanismPlain
+	SASLMechanismScramSHA256            SASLMechanism = config.SASLMechanismScramSHA256
+	SASLMechanismScramSHA512            SASLMechanism = config.SASLMechanismScramSHA512
+	SASLMechanismGSSAPI                 SASLMechanism = config.SASLMechanismGSSAPI
+	SASLMechanismOAuthBearer            SASLMechanism = config.SASLMechanismOAuthBearer
+	SASLMechanismAWSManagedStreamingIAM SASLMechanism = config.SASLMechanismAWSManagedStreamingIAM
+)
+
+// KafkaSASLOAuthBearer is the config struct for the SASL OAuthBearer mechanism
+type KafkaSASLOAuthBearer struct {
+	Token SecretKeyRef `json:"tokenSecretRef"`
+}
+
+// KafkaSASLGSSAPI represents the Kafka Kerberos config.
+type KafkaSASLGSSAPI struct {
+	AuthType           string       `json:"authType"`
+	KeyTabPath         string       `json:"keyTabPath"`
+	KerberosConfigPath string       `json:"kerberosConfigPath"`
+	ServiceName        string       `json:"serviceName"`
+	Username           string       `json:"username"`
+	Password           SecretKeyRef `json:"passwordSecretRef"`
+	Realm              string       `json:"realm"`
+
+	// EnableFAST enables FAST, which is a pre-authentication framework for Kerberos.
+	// It includes a mechanism for tunneling pre-authentication exchanges using armored KDC messages.
+	// FAST provides increased resistance to passive password guessing attacks.
+	EnableFast bool `json:"enableFast"`
+}
+
+// KafkaSASLAWSMskIam is the config for AWS IAM SASL mechanism,
+// see: https://docs.aws.amazon.com/msk/latest/developerguide/iam-access-control.html
+type KafkaSASLAWSMskIam struct {
```

Comment:
nit: can we be consistent about how acryonyms are handled in the naming here? either `KafkaSASLAWSMSKIAM` or `KafkaSaslAwsMskIam`. Not that either are particularly nice to look at.

Ah this is just being moved, ignore me.